src/main/java/seedu/internbuddy/model/company/Company.java

```diff
@@ -22,15 +25,17 @@ public class Company {
     private final Email email;
 
     // Data fields
+    private final ArrayList<Application> applications;
     private final Address address;
-    private final Status status;
+    private Status status;
     private final Set<Tag> tags = new HashSet<>();
 
     /**
      * Every field must be present and not null.
      */
     public Company(Name name, Phone phone, Email email, Address address, Set<Tag> tags, Status status) {
         requireAllNonNull(name, phone, email, address, tags, status);
+        this.applications = new ArrayList<>();
```

Comment:
I think adding `applications` to the constructor would be appropriate in this case since `tags` is also in the constructor.